modules/consensus/synchronize_test.go

```diff
@@ -311,3 +316,488 @@ func TestSendBlocksBroadcastsOnce(t *testing.T) {
+				expectedID := types.BlockID{}
+				if id != expectedID {
+					t.Errorf("id written to conn was %v, but id received was %v", expectedID, id)
+				}
+				fnDone <- struct{}{}
+			},
+			errWant: errFailingReader,
+			msg:     "the function returned from threadedReceiveBlock should fail with a PeerConn with a failing reader",
+		},
+		// Test with a valid conn, but an invalid block.
+		{
+			id:   types.BlockID{1},
+			conn: p1,
+			fn: func() {
+				// Read the id written to conn.
+				var id types.BlockID
+				if err := encoding.ReadObject(p2, &id, crypto.HashSize); err != nil {
+					t.Error(err)
+				}
+				// Verify the id is the expected id.
+				expectedID := types.BlockID{1}
+				if id != expectedID {
+					t.Errorf("id written to conn was %v, but id received was %v", expectedID, id)
+				}
+
+				// Write an invalid block to conn.
+				block := types.Block{}
+				if err := encoding.WriteObject(p2, block); err != nil {
+					t.Error(err)
+				}
+
+				fnDone <- struct{}{}
+			},
+			errWant: errOrphan,
+			msg:     "the function returned from threadedReceiveBlock should not accept an invalid block",
+		},
+		// Test with a valid conn and a valid block.
+		{
+			id:   types.BlockID{2},
+			conn: p1,
+			fn: func() {
+				// Read the id written to conn.
+				var id types.BlockID
+				if err := encoding.ReadObject(p2, &id, crypto.HashSize); err != nil {
+					t.Error(err)
+				}
+				// Verify the id is the expected id.
+				expectedID := types.BlockID{2}
+				if id != expectedID {
+					t.Errorf("id written to conn was %v, but id received was %v", expectedID, id)
+				}
+
+				// Write a valid block to conn.
+				block, err := cst.miner.FindBlock()
+				if err != nil {
+					t.Error(err)
+				}
+				if err := encoding.WriteObject(p2, block); err != nil {
+					t.Error(err)
+				}
+
+				fnDone <- struct{}{}
+			},
+			errWant: nil,
+			msg:     "the function returned from manageddReceiveBlock should accept a valid block",
+		},
+	}
+	for _, tt := range tests {
+		managedReceiveFN := cst.cs.threadedReceiveBlock(tt.id)
+		go tt.fn()
+		err := managedReceiveFN(tt.conn)
+		if err != tt.errWant {
+			t.Errorf("%s: expected to fail with `%v', got: `%v'", tt.msg, tt.errWant, err)
+		}
+		<-fnDone
+	}
+}
+
+// TestIntegrationSend1BlkRPC probes the Send1Blk RPC and tests that blocks are
+// correctly requested, received, and accepted into the consensus set.
+func TestIntegrationSend1BlkRPC(t *testing.T) {
+	cst1, err := blankConsensusSetTester("TestIntegrationSend1BlkRPC1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst1.Close()
+	cst2, err := blankConsensusSetTester("TestIntegrationSend1BlkRPC2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst2.Close()
+
+	err = cst1.cs.gateway.Connect(cst2.cs.gateway.Address())
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = cst2.cs.gateway.Connect(cst1.cs.gateway.Address()) // TODO: why is this Connect call necessary? Shouldn't they connect to eachother with one Connect?
```

Comment:
does it have something to do with inbound/outbound peers, or some level of mocking that's happening somewhere? Perhaps it's a bug.